### PR TITLE
MTL-1800 daemons

### PR DESCRIPTION
**Pull request by rustydb**
#### Summary and Scope

- Fixes MTL-1800

##### Issue Type

- Bugfix Pull Request

This pulls in a new metal-provision that has better daemon handling for NCN common and NCN metal.

#### Prerequisites

- [x] I have included documentation in my PR (or it is not required)
- [x] I tested this on metal (e.g. an internal system, with hardware) (x) (if yes, please include results or a description of the test)
- [x] I tested this on vshasta (if yes, please include results or a description of the test)

### CODE REVIEW

File: vendor/github.com/Cray-HPE/metal-provision/roles/ncn-common-install/tasks/common.yml [hunk 1]
```diff
@@ -1,114 +1,72 @@
...
+- name: Setup Daemons
   systemd:
-    name: "{{ item }}"
-    enabled: no
-    state: stopped
-  with_items:
-    - postfix.service
+    enabled: "{{ item.enabled }}"
+    name: "{{ item.name }}"
+    masked: "{{ item.masked | default(no) }}"
```
[the remark this answers is not in the document]
Comment (rustydb) [reply vendor/github.com/Cray-HPE/metal-provision/roles/ncn-common-install/tasks/common.yml:70]: Good catch, I'll fix that in https://github.com/Cray-HPE/metal-provision/pull/5

[the remark this answers is not in the document]
Comment (rustydb) [reply vendor/github.com/Cray-HPE/metal-provision/roles/ncn-common-install/tasks/common.yml:70]: I think actually `masked:` takes an Ansible/YAML boolean.. so it should be `no` sans quotes.

[the remark this answers is not in the document]
Comment (rustydb) [reply vendor/github.com/Cray-HPE/metal-provision/roles/ncn-common-install/tasks/common.yml:70]: We'll see how the current build goes.

[the remark this answers is not in the document]
Comment (rustydb) [reply vendor/github.com/Cray-HPE/metal-provision/roles/ncn-common-install/tasks/common.yml]: Needed to be `false`. `'no'` might've worked, but the issue with `no` is that it doesn't translate to a boolean in the Jinja syntax, so it wasn't a valid value. Ansible was complaining that `no` wasn't defined. `false` works, and doesn't need quotes.

**Review by heemstra — APPROVED** [review]

**Review by heemstra — APPROVED** [review]

**Pull request merged**